site/src/templates.rs

```diff
@@ -0,0 +1,37 @@
+use parking_lot::RwLock;
+use std::path::{Path, PathBuf};
+use tera::Tera;
+
+pub struct TemplateRenderer {
+    template_dir: PathBuf,
+    tera: RwLock<Tera>,
```

Comment:
Just want to check that we're fine using a sync lock here in async contexts. I think this should be fine since live reloading should only be used locally where the lock shouldn't be under contention.

Reply:
That's a good point. But indeed as you said, on production environments it should never even block, as there would be only readers. I can make it async though if you'd like, it's a small change.

Reply:
If it's a small change, it might be worth to do, but I don't think it really matters.

Reply:
I checked the rest of the code and this function anyway did blocking file I/O... so maybe it would require a larger refactor so that it would make sense. I can change it to async, but then there would still be sync code just next to it.